### DISCUSSION TITLE
Cannot change default theme by session

### DISCUSSION THREAD

**Discussion by damnh** [primefaces/community#174] [PrimeReact Templates]
I save themes in session (next-auth). So when I use getServerSession to get theme from session to edit app/layout file from href={/theme/theme-light/green/theme.css} to href={/theme/theme-light/${user.themes}/theme.css}, the page is loaded infinitely, when I checked console, the error is following. Can you show me where is wrong? Thank you!

**This is error:**

`Uncaught (in promise) TypeError: Cannot destructure property 'children' of 'param' as it is undefined.
    at RootLayout (webpack-internal:///(app-client)/./app/layout.tsx:29:11)
    at describeNativeComponentFrame (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:2534:7)
    at describeFunctionComponentFrame (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:2629:12)
    at describeFiber (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:2708:14)
    at getStackByFiberInDevAndProd (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:2727:15)
    at createCapturedValueAtFiber (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:14060:12)
    at throwException (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:14550:11)
    at throwAndUnwindWorkLoop (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24676:5)
    at renderRootConcurrent (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24291:15)
    at performConcurrentWorkOnRoot (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:23343:38)
    at workLoop (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:261:34)
    at flushWork (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:230:14)
    at MessagePort.performWorkUntilDeadline (webpack-internal:///(app-client)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:537:21)
RootLayout @ layout.tsx:19
describeNativeComponentFrame @ react-dom.development.js:2534
describeFunctionComponentFrame @ react-dom.development.js:2629
describeFiber @ react-dom.development.js:2708
getStackByFiberInDevAndProd @ react-dom.development.js:2727
createCapturedValueAtFiber @ react-dom.development.js:14060
throwException @ react-dom.development.js:14550
throwAndUnwindWorkLoop @ react-dom.development.js:24676
renderRootConcurrent @ react-dom.development.js:24291
performConcurrentWorkOnRoot @ react-dom.development.js:23343
workLoop @ scheduler.development.js:261
flushWork @ scheduler.development.js:230
performWorkUntilDeadline @ scheduler.development.js:537
640app-index.js:31 Warning: An error occurred during hydration. The server HTML was replaced with client content in <#document>.
window.console.error @ app-index.js:31
console.error @ hydration-error-info.js:45
printWarning @ react-dom.development.js:94
error @ react-dom.development.js:68
errorHydratingContainer @ react-dom.development.js:34648
recoverFromConcurrentError @ react-dom.development.js:23446
performConcurrentWorkOnRoot @ react-dom.development.js:23356
workLoop @ scheduler.development.js:261
flushWork @ scheduler.development.js:230
performWorkUntilDeadline @ scheduler.development.js:537
login:1 Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received
23706app-index.js:31 Warning: An error occurred during hydration. The server HTML was replaced with client content in <#document>.`

**Comment by habubey** [comment]
Which template exactly are you working on? and Which version?

**Reply by damnh** [reply]
Hi, I have just updated the Apollo v10.
Thanks

**Reply by habubey** [reply]
It is hard to debug without your codes. The config works well currently in Apollo. But it is probably about your RootLayout. 
Are you sure that param is an object, and then destructure the prop of the children of the param object?

**Reply by damnh** [reply]
1. I using next-auth to connect DB to authenticate and login, in callback, I set theme into session.


`
    export const authOptions: NextAuthOptions = {

    providers: [
        CredentialsProvider({
        name: "Credentials",
        credentials: {
            username: { label: "Username", type: "text", placeholder: "username" },
            password: { label: "Password", type: "password" },
        },
        async authorize(credentials, req) {
            // login by our Authenticate, call rest API to signin
            if (user) {
                return user;
            } else {
                return null;
            }
        },
        }),
    ],
    callbacks: {
        async jwt({ token, user, trigger, session }) {
            if (trigger === "update") {
                return { ...token, ...session.user };
            }
            return { ...token, ...user };
        },
        async session({ session, token }) {
            session.user = token as any;
            return session;
        },
    },
    session: {
        strategy: 'jwt',
        maxAge: 30*24*60*60,
    },
    pages : {
        signIn: '/auth/login'
    }
 };

  export const handler = NextAuth(authOptions);
  export { handler as DELETE, handler as GET, handler as POST, handler as PUT };
`


2. Using login page of Apollo (in demo/(full-page))
3. After login, page is redirected to locahost:3000/. At this time I want to change theme by session's saved theme.
4. I have customized file app/layout.tsx to get theme


`
       export default async function RootLayout({ children }: RootLayoutProps) {  
       const session = await getServerSession(authOptions);
       let userInfo: UserInfo = getLoginUser(session)!;

    return (

    <html lang="en" suppressHydrationWarning>
      <head>
        <link
          id="theme-link"
          href={`/theme/theme-light/${userInfo.themes}/theme.css`}
          rel="stylesheet"
        ></link>
      </head>
      <body>
        <PrimeReactProvider>
          <LayoutProvider>{children}</LayoutProvider>
        </PrimeReactProvider>
      </body>
    </html>

     );

     }
`

5. I want share session, so I've editted the layout.tsx of (main) in this following:

`
    export default function MainLayout({ children }: MainLayoutProps) {

        const session = await getServerSession(authOptions);
         return (

            <SessionProvider session={session}>
                <Layout>{children}</Layout>
            </SessionProvider>

        );
    }
`

Now I test on (full-page) and (main) folder of Apollo.
Thanks!

**Reply by damnh** [reply]
Hi @habubey 
Could you please support me about this problem?
Thanks!

**Reply by habubey** [reply]
Hi! We have put it in the queue, it may be a general issue, but we will get back to you.

**Reply by mertsincan** [reply]
So sorry for the delayed response! @nitrogenous can you assist with this ticket?
